lib/Dialect/StandardOps/Ops.cpp

```diff
@@ -2021,6 +2021,24 @@ OpFoldResult SelectOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// SignExtendIOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verify(SignExtendIOp op) {
+  auto opType = op.getOperand()->getType().dyn_cast_or_null<IntegerType>();
+  if (!opType)
+    return op.emitError("operand type ") << opType << " is not an integer ";
```

Comment:
Because you use a constraint on the operand in ODS, there is already a verifier generated for you.
And the actual verifier is accepting Tensors and Vector (you can change the constraint to limit to only scalar if desirable).

You can always check the generated code by running:

`{build_path}/bin/mlir-tblgen -gen-op-defs {src_path}/include/mlir/Dialect/StandardOps/Ops.td -I {src_path}/include/ `

Reply:
Thanks for this hint. Now I understood, how the verifiers are chained and which checks are already auto-generated.